### PR TITLE
fix(route): fix pianyuan, add  pianyuan/search

### PR DESCRIPTION
**Pull request by jerry1119**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/pianyuan/index
/pianyuan/index/mv
/pianyuan/index/tv
/pianyuan/indexers/pianyuan/results/search/
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
更新到片源网新地址
现在只能通过coockie来访问
规范rss输出格式,增加磁链输出
新增搜索路由,改动原路由地址

### CODE REVIEW

File: lib/routes/pianyuan/utils.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const security_key = 'pianyuan-security_session_verify';
+const PHPSESSID_key = 'pianyuan-PHPSESSID';
+const loginauth_key = 'pianyuan-py_loginauth';
+
+const ProcessFeed = async (list, cache) => {
+    const link_base = 'https://pianyuan.org';
+
+    const items = await Promise.all(
+        list.map(async (e) => {
+            const link = url.resolve(link_base, e);
```
Comment (TonyRL) [inline lib/routes/pianyuan/utils.js:14]: `url.resolve` is deprecated. Please use [`URL`](https://nodejs.org/api/url.html#the-whatwg-url-api) instead.

File: lib/routes/pianyuan/utils.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+                return {
+                    title: `${torrent_name} [${name}] [${size}]`,
+                    size: length,
+                    enclosure_url: magnet,
+                    enclosure_type: 'application/x-bittorrent',
+                    enclosure_length: length,
+                };
+            });
+            return Promise.resolve(single);
```
Comment (TonyRL) [inline lib/routes/pianyuan/utils.js:38]:
```suggestion
            return single;
```

File: lib/routes/pianyuan/search.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        searchLinks.map(async (e) => {
+            const link = url.resolve(link_base, e);
+            const single = await ctx.cache.tryGet(link, async () => {
+                const res = await utils.request(link, ctx.cache);
+                const content = cheerio.load(res.data);
+                content('.ico.ico_bt')
+                    .get()
+                    .map((a) => detailLinks.push($(a).attr('href')));
+            });
+            return Promise.resolve(single);
```
Comment (TonyRL) [inline lib/routes/pianyuan/search.js:34]:
```suggestion
            return single;
```

**Comment by TonyRL** [comment]
The issues mentioned above have not been resolved yet.

File: lib/routes/pianyuan/utils.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    return items;
+};
+
+async function getCookie(cache) {
+    const security_session_verify = await cache.get(security_key);
+    const PHPSESSID = await cache.get(PHPSESSID_key);
+    let py_loginauth = await cache.get(loginauth_key);
+    if (!py_loginauth) {
+        if (!config.pianyuan || !config.pianyuan.cookie) {
+            throw 'pianyuan  is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
```
Comment (Rongronggg9) [inline lib/routes/pianyuan/utils.js:50]:
```suggestion
            throw 'pianyuan is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
```

**Review by Rongronggg9 — CHANGES_REQUESTED** [review]

File: docs/multimedia.md [hunk 5]
```diff
@@ -1260,9 +1260,25 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
...
+::: tip 片源网已经关闭公开访问
+
+片源信息为登陆后可见，自建时将环境变量`PIANYUAN_COOKIE`设为登陆后的 Cookie 中的 py_loginauth 值，即可正常获取。
+例: PIANYUAN_COOKIE='py_loginauth=xxxxxxxxxx'
+:::
+
 ### 最新资源
 
-<Route author="greatcodeeer" example="/pianyuan" path="/pianyuan" radar="1" rssbud="1"/>
+<Route author="greatcodeeer jerry1119" example="/pianyuan/index" path="/pianyuan/index/:media?" :paramsDesc="['类别，见下表，默认为首页']" radar="1" rssbud="1"/>
```
Comment (TonyRL) [inline docs/multimedia.md:1271]:
```suggestion
<Route author="greatcodeeer jerry1119" example="/pianyuan/index" path="/pianyuan/index/:media?" :paramsDesc="['类别，见下表，默认为首页']" radar="1" rssbud="1"  selfhost="1"/>
```

File: docs/multimedia.md [hunk 6]
```diff
@@ -1260,9 +1260,25 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
...
+
+| 电影 | 剧集 |
+| -- | -- |
+| mv | tv |
+
+### 搜索
+
+<Route author="jerry1119" example="/pianyuan/indexers/pianyuan/results/search/api?t=test&q=长津湖" path="/pianyuan/indexers/pianyuan/results/search/" />
```
Comment (TonyRL) [inline docs/multimedia.md:1279]:
```suggestion
<Route author="jerry1119" example="/pianyuan/indexers/pianyuan/results/search/api?t=test&q=长津湖" path="/pianyuan/indexers/pianyuan/results/search/" selfhost="1"/>
```

File: docs/multimedia.md [hunk 7]
```diff
@@ -1260,9 +1260,25 @@ JavDB 有多个备用域名，本路由默认使用永久域名 <https://javdb.c
 
 ## 片源网
 
+::: tip 片源网已经关闭公开访问
+
+片源信息为登陆后可见，自建时将环境变量`PIANYUAN_COOKIE`设为登陆后的 Cookie 中的 py_loginauth 值，即可正常获取。
+例: PIANYUAN_COOKIE='py_loginauth=xxxxxxxxxx'
+:::
```
Comment (TonyRL) [inline docs/multimedia.md:1267]: Please add this to `docs/install/README.md` instead

File: lib/routes/pianyuan/app.js [hunk 8]
```diff
@@ -17,28 +17,13 @@ module.exports = async (ctx) => {
...
+    const response = await utils.request(link, ctx.cache);
+    const $ = cheerio.load(response.data);
+    const detailLinks = $('#main-container > div > div.col-md-10 > table > tbody > tr')
+        .get()
+        .map((tr) => $(tr).find('td.dt.prel.nobr > a').attr('href'));
+    detailLinks.shift();
+    const items = await utils.ProcessFeed(detailLinks, ctx.cache);
 
     ctx.state.data = {
         title: `片源网`,
```
Comment (TonyRL) [inline lib/routes/pianyuan/app.js:29]:
```suggestion
        title: '片源网',
```

File: lib/routes/pianyuan/search.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    const link = link_base + `search?q=${searchKey}`;
+
+    const response = await utils.request(link, ctx.cache);
+    const $ = cheerio.load(response.data);
+    // 只获取第一页的搜索结果
+    const searchLinks = $('.nomt > a')
+        .get()
+        .map((a) => $(a).attr('href'));
+    if (searchLinks.length === 0) {
+        throw `pianyuan 搜索失败`;
```
Comment (TonyRL) [inline lib/routes/pianyuan/search.js:18]:
```suggestion
        throw 'pianyuan 搜索失败';
```

File: lib/routes/pianyuan/search.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+                    .map((a) => detailLinks.push($(a).attr('href')));
+            });
+            return single;
+        })
+    );
+
+    const items = await utils.ProcessFeed(detailLinks, ctx.cache);
+
+    ctx.state.data = {
+        title: `片源网`,
```
Comment (TonyRL) [inline lib/routes/pianyuan/search.js:40]:
```suggestion
        title: '片源网',
```

File: lib/routes/pianyuan/utils.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+                cache.set(security_key, e.split(';')[0]);
+            } else if (e.indexOf('PHPSESSID') === 0) {
+                cache.set(PHPSESSID_key, e.split(';')[0]);
+            } else if (e.indexOf('py_loginauth') === 0) {
+                cache.set(loginauth_key, e.split(';')[0]);
+            }
+        }
+    }
+    if (response.data.includes('会员登录后才能访问')) {
+        throw `pianyuan Cookie已失效`;
```
Comment (TonyRL) [inline lib/routes/pianyuan/utils.js:80]:
```suggestion
        throw 'pianyuan Cookie已失效';
```

**Pull request merged**